### PR TITLE
Remove useless condition and add more check

### PR DESCRIPTION
**Pull request by gnomeby**
* Add more check of ROM reading
* Remove never True condition

### CODE REVIEW

**Comment by kervinck** [comment]
Thanks! I have two remarks on this proposal:

1. For it to be useful as instruction set definition, gtemu.c must fit on the 2 pages in the printed manual. That's why there's no additional checking, and all code is a bit concise. If the ROM size is smaller, that's not a problem by itself, I think? I frequently make ROM files that are not filled, for example for my breadboard version of the Gigatron.

2. In my view, the vgaX==200 condition is still important for testing ROM changes. The graphical emulators are very bad at detecting cycle count errors, while the VGA signal must be cycle perfect for it to work on many physical monitors. I normally usethe 'make test' target tests for the removed code. The '>' indicator is then very handy to know what kind of bug we're dealing with.

Of course, any emulator that is derived from gtemu.c can add whatever feature it wants.

**Comment by Cwiiis** [comment]
Re. 2, the `print '>'` won't ever be hit because of the if condition wrapping the increment of vgaX - but I suppose the correct fix is to always increment, then have an `if (vgaX < 200) { ... } else if (vgaX == 200) { print '>' }`

**Comment by gnomeby** [comment]
Here you are.

File: Docs/gtemu.c [hunk 1]
```diff
@@ -106,7 +106,7 @@ int main(int argc, char *argv[])
     int hSync = (T.OUT & 0x40) - (S.OUT & 0x40); // "VGA monitor" (use simple stdout)
     int vSync = (T.OUT & 0x80) - (S.OUT & 0x80);
     if (vSync < 0) vgaY = -36; // Falling vSync edge
-    if (vgaX++ < 200) {
+    if (vgaX++ <= 200) {
       if (hSync) putchar('|');              // Visual indicator of hSync
       else if (vgaX == 200) putchar('>');   // Too many pixels
```
Comment (Cwiiis) [inline Docs/gtemu.c]: Maybe Marcel can comment, but I wonder if the vgaX == 200 check should be first before the hSync check?

Comment (kervinck) [reply Docs/gtemu.c]: The intent of the code is the following:

1. Scanlines emit characters up to and including 200 cycles (0..199)
2. Scanlines _longer_ than 200 cycles get truncated with '>'
3. Scanlines _different_ from 200 cycles get marked with '~'

Indeed, vgaX won't count beyond 200, but that's not needed for the above. It _does_ reach 200 however. vgaX++ is post-increment.

Comment (gnomeby) [reply Docs/gtemu.c]: Now I understand you. The original code is correct.

**Pull request closed without merging**